src/app/source-file-tabs/source-file-tabs.component.html

```diff
@@ -32,16 +32,51 @@
         <div *ngIf="fileTab.value.length > 0">
           <mat-toolbar color="primary">
             <div class="w-100" fxLayout="row" fxLayoutAlign="space-between center">
-              <button color="accent" *ngIf="fileTab.value.length > 10" mat-raised-button (click)="openFileTree(fileTab.value)">
-                Select File
-              </button>
+              <span class="select-file-button-span" *ngIf="fileTab.value.length > 10">
+                <button color="accent" data-cy="select-file-button" mat-raised-button (click)="openFileTree(fileTab.value)">
+                  Select File
+                </button>
+              </span>
               <span class="no-wrap" *ngIf="fileTab.value.length > 10"> {{ fileName }} </span>
+              <span
+                *ngIf="isCurrentFilePrimary && fileTab.value.length > 10"
+                class="bubble primary-bubble primary-bubble-header"
+                data-cy="primary-descriptor-bubble"
+              >
+                Primary
+              </span>
+
               <mat-form-field *ngIf="fileTab.value.length <= 10" class="w-50">
                 <mat-select [value]="currentFile" (selectionChange)="matSelectChange($event)">
-                  <mat-option [value]="file" *ngFor="let file of fileTab.value"> {{ file.path }} </mat-option>
+                  <mat-select-trigger [value]="currentFile">
```

Comment:
My IDE says `[value]` isn't part of the `mat-select-trigger` directive -- do you need it?